### PR TITLE
Initial hwloc-devel

### PR DESCRIPTION
**Pull request by michalbiesek**
### Types of changes

- [ ] Bugfix (non-breaking change which fixes issue linked in Description above)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (correction or otherwise)
- [ ] Cosmetics (whitespace, appearance)
- [ ] Other

### Checklist

- [ ] Code compiles without errors
- [ ] All tests pass locally with my changes (see TESTING section in CONTRIBUTING file)
- [ ] Created tests which will fail without the change (if possible)
- [ ] Extended the README/documentation (if necessary)
- [ ] All newly added files have proprietary license (if necessary)
- [ ] All newly added files are referenced in MANIFEST files (if necessary)

## Further comments

---
This change is [Reviewable](https://reviewable.io/reviews/memkind/memkind/389)

### CODE REVIEW

File: README [hunk 1]
```diff
@@ -74,6 +74,10 @@ You will need to install required packages on the build system:
 * **numactl-devel**
 * **unzip**
 
+For using HMAT features in memkind:
```
Comment (jschmieg) [inline README:77]: I suggest adding a short description when "HMAT" appears the first time. What it gives to memkind and happens if it is not present.

**Review by PatKamin — COMMENTED** [review]
> Reviewed 10 of 10 files at r1.
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/389)* status: all files reviewed, 5 unresolved discussions (waiting on @michalbiesek)

___
*[configure.ac, line 220 at r1](https://reviewable.io/reviews/memkind/memkind/389#-MKE68Pg4tAmbVfxeAQl:-MKE68Pg4tAmbVfxeAQm:b1s6zbx) ([raw file](https://github.com/memkind/memkind/blob/812a92f562e2d727a0d8ce5572a90620e076eff1/configure.ac#L220)):*
> ```M4Sugar
> fi
> if test "x$enable_hwloc" = "x2" ; then
>     AC_CHECK_LIB(hwloc, hwloc_memattr_get_best_target, [hwloc_hmat=yes], [AC_MSG_ERROR([libhwloc-devel v 2.3.0 or later is required dependency by --enable-hwloc option])])
> ```

nit: v2.3.0
___
*[configure.ac, line 223 at r1](https://reviewable.io/reviews/memkind/memkind/389#-MKE7Y6YDgAWyxHz0lWX:-MKE7Y6YDgAWyxHz0lWY:b-8pjtm7) ([raw file](https://github.com/memkind/memkind/blob/812a92f562e2d727a0d8ce5572a90620e076eff1/configure.ac#L223)):*
> ```M4Sugar
> fi
> 
> if test $hwloc_enable = "yes" ; then
> ```

`$hwloc_hmat`
___
*[configure.ac, line 275 at r1](https://reviewable.io/reviews/memkind/memkind/389#-MKEA3Re4Yf-E1iZaqau:-MKEA3Re4Yf-E1iZaqav:b1s6zbx) ([raw file](https://github.com/memkind/memkind/blob/812a92f562e2d727a0d8ce5572a90620e076eff1/configure.ac#L275)):*
> ```M4Sugar
> AC_MSG_RESULT([
>                Utilize information about memory performance characteristics is disabled
>                To enable this feature please install libhwloc-devel v 2.3.0 or later
> ```

nit: v2.3.0
___
*[utils/docker/docker_install_hwloc.sh, line 28 at r1](https://reviewable.io/reviews/memkind/memkind/389#-MKECb7P2JzengL-tkpo:-MKECb7P2JzengL-tkpp:boed5tm) ([raw file](https://github.com/memkind/memkind/blob/812a92f562e2d727a0d8ce5572a90620e076eff1/utils/docker/docker_install_hwloc.sh#L28)):*
> ```Shell
> cd "$HWLOC_LOCAL_DIR"
> ./configure --prefix=/usr
> make -j "$(nproc --all)"
> ```

Worth considering: Remove the `--all` argument to use maximum number of cores **available** to the process.
Example: On a machine that runs with 48 physical cores and with Hyper-Threading turned off, you will have available 48 cores. However, `nproc --all` will output 96. I don't know whether `make` limits runned jobs somehow, but it seems that you will run 96 jobs on 48 cores in this case when using `--all`.
Same below.

**Review by michalbiesek — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/389)* status: 7 of 10 files reviewed, 4 unresolved discussions (waiting on @jschmieg and @PatKamin)

___
*[configure.ac, line 220 at r1](https://reviewable.io/reviews/memkind/memkind/389#-MKE68Pg4tAmbVfxeAQl:-MKEkM-a5pMY-RdY9eoF:bb6un1j) ([raw file](https://github.com/memkind/memkind/blob/812a92f562e2d727a0d8ce5572a90620e076eff1/configure.ac#L220)):*
<details><summary><i>Previously, PatKamin (Patryk Kaminski) wrote…</i></summary><blockquote>

nit: v2.3.0
</blockquote></details>

Done
___
*[configure.ac, line 223 at r1](https://reviewable.io/reviews/memkind/memkind/389#-MKE7Y6YDgAWyxHz0lWX:-MKEkbN62iiSNFCT8unf:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/812a92f562e2d727a0d8ce5572a90620e076eff1/configure.ac#L223)):*
<details><summary><i>Previously, PatKamin (Patryk Kaminski) wrote…</i></summary><blockquote>

`$hwloc_hmat`
</blockquote></details>

Done.
___
*[configure.ac, line 275 at r1](https://reviewable.io/reviews/memkind/memkind/389#-MKEA3Re4Yf-E1iZaqau:-MKEkcbX5pjkk4cCon2M:bb6un1j) ([raw file](https://github.com/memkind/memkind/blob/812a92f562e2d727a0d8ce5572a90620e076eff1/configure.ac#L275)):*
<details><summary><i>Previously, PatKamin (Patryk Kaminski) wrote…</i></summary><blockquote>

nit: v2.3.0
</blockquote></details>

Done
___
*[README, line 77 at r1](https://reviewable.io/reviews/memkind/memkind/389#-MK9z9fNS_GR2VIlLoSN-r1-77:-MKEkem6714PO51ZzMg1:b-pxf8ao) ([raw file](https://github.com/memkind/memkind/blob/812a92f562e2d727a0d8ce5572a90620e076eff1/README#L77)):*
<details><summary><i>Previously, jschmieg wrote…</i></summary><blockquote>

I suggest adding a short description when "HMAT" appears the first time. What it gives to memkind and happens if it is not present.
</blockquote></details>

Done with the section below (this could be updated later when we have new memory kinds added)
___
*[utils/docker/docker_install_hwloc.sh, line 28 at r1](https://reviewable.io/reviews/memkind/memkind/389#-MKECb7P2JzengL-tkpo:-MKEkm2M0ZzLWkK7PoBR:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/812a92f562e2d727a0d8ce5572a90620e076eff1/utils/docker/docker_install_hwloc.sh#L28)):*
<details><summary><i>Previously, PatKamin (Patryk Kaminski) wrote…</i></summary><blockquote>

Worth considering: Remove the `--all` argument to use maximum number of cores **available** to the process.
Example: On a machine that runs with 48 physical cores and with Hyper-Threading turned off, you will have available 48 cores. However, `nproc --all` will output 96. I don't know whether `make` limits runned jobs somehow, but it seems that you will run 96 jobs on 48 cores in this case when using `--all`.
Same below. 
</blockquote></details>

Done.

**Review by PatKamin — COMMENTED** [review]
> Reviewed 4 of 4 files at r2.
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/389)* status: all files reviewed, 2 unresolved discussions (waiting on @jschmieg and @michalbiesek)

___
*[README, line 212 at r2](https://reviewable.io/reviews/memkind/memkind/389#-MKEo0Px8bnfyLsZd9b_:-MKEo0Py9ZcaW-2jLb2v:b-sr9675) ([raw file](https://github.com/memkind/memkind/blob/7e7a42ca6d1b55b1fa38cd45a2dffb466b50d9fd/README#L212)):*
> ```
> 
> To utilize memory performance characteristics (latency and bandwidth),
> besides [platform](https://uefi.org/sites/default/files/resources/ACPI_6_3_final_Jan30.pdf)
> ```

beside

**Review by michalbiesek — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/389)* status: 9 of 10 files reviewed, 2 unresolved discussions (waiting on @jschmieg and @PatKamin)

___
*[README, line 212 at r2](https://reviewable.io/reviews/memkind/memkind/389#-MKEo0Px8bnfyLsZd9b_:-MKEskym1l_VeQm-gp6N:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/7e7a42ca6d1b55b1fa38cd45a2dffb466b50d9fd/README#L212)):*
<details><summary><i>Previously, PatKamin (Patryk Kaminski) wrote…</i></summary><blockquote>

beside
</blockquote></details>

Done.

**Review by PatKamin — COMMENTED** [review]
> Reviewed 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/389)* status: all files reviewed, 1 unresolved discussion (waiting on @jschmieg)

**Review by PatKamin — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/389#-:-MKEtpCt74dN2ZxL_0hO:bnfp4nl)* status: all files reviewed, 1 unresolved discussion (waiting on @jschmieg)

**Pull request merged**